hca/dss/__init__.py

```diff
@@ -255,19 +261,22 @@ def upload(self, src_dir, replica, staging_bucket, timeout_seconds=1200):
         version = datetime.utcnow().strftime("%Y-%m-%dT%H%M%S.%fZ")
 
         files_to_upload, files_uploaded = [], []
-        for filename in os.listdir(src_dir):
-            full_file_name = os.path.join(src_dir, filename)
+        for filename in directory_builder(src_dir):
+            full_file_name = filename.path
             files_to_upload.append(open(full_file_name, "rb"))
 
         logger.info("Uploading %i files from %s to %s", len(files_to_upload), src_dir, staging_bucket)
-        file_uuids, uploaded_keys = upload_to_cloud(files_to_upload, staging_bucket=staging_bucket, replica=replica,
-                                                    from_cloud=False)
+        file_uuids, uploaded_keys, abs_file_paths = upload_to_cloud(files_to_upload, staging_bucket=staging_bucket,
+                                                                    replica=replica, from_cloud=False)
         for file_handle in files_to_upload:
             file_handle.close()
-        filenames = list(map(os.path.basename, uploaded_keys))
+        filenames = [object_name_builder(p, src_dir) for p in abs_file_paths]
         filename_key_list = list(zip(filenames, file_uuids, uploaded_keys))
 
         for filename, file_uuid, key in filename_key_list:
+            filename = filename.replace('\\', '/')  # for windows paths
```

Comment:
Does this work with `C:\\path\path\file.ex`?  I think it might need to strip the drive name for Windows files.

Reply:
the upload command would not work in this case, as it's only designated for directories, however it does handle the case in which the drive letter is included; within a support function the lettering for the directory is stripped as needed, see [hca/dss/util/__init__.py object_name_builder](hca/dss/util/__init__.py)